service/header/sync.go

```diff
@@ -14,24 +16,35 @@ type Syncer struct {
 	store    Store
 	trusted  tmbytes.HexBytes
 
-	// done is triggered when syncing is finished
-	// it assumes that Syncer is only used once
-	done chan struct{}
+	// inProgress is set to 1 once syncing commences and
+	// is set to 0 once syncing is either finished or
+	// not currently in progress
+	inProgress *uint64
```

Comment:
Atomic values should be stored on top of the strcut def, because of https://golang.org/pkg/sync/atomic/#pkg-note-BUG. Also, you should store it not as a pointer but as a value, then you need to do atomic operations by referencing values e.g. `atomic.StoreUint64(&s.inProgress, 1)`